lib/form/sfFormField.class.php

```diff
@@ -35,13 +35,13 @@ class sfFormField
     /**
      * Constructor.
      *
-     * @param sfWidgetForm     $widget A sfWidget instance
-     * @param sfFormField      $parent The sfFormField parent instance (null for the root widget)
-     * @param string           $name   The field name
-     * @param string           $value  The field value
-     * @param sfValidatorError $error  A sfValidatorError instance
+     * @param sfWidgetForm          $widget A sfWidget instance
+     * @param sfFormField|null      $parent The sfFormField parent instance (null for the root widget)
+     * @param string                $name   The field name
+     * @param mixed                 $value  The field value
+     * @param sfValidatorError|null $error  A sfValidatorError instance
      */
-    public function __construct(sfWidgetForm $widget, ?sfFormField $parent = null, $name, $value, ?sfValidatorError $error = null)
+    public function __construct(sfWidgetForm $widget, ?sfFormField $parent = null, $name = '', $value = null, ?sfValidatorError $error = null)
```

Comment:
The reason `$parent` default value was `null` is because when argument types were introduced in PHP 5.4, the only way to make it nullable was to put the `null` default value. https://durak.org/sean/pubs/software/php-5.4.6/language.oop5.typehinting.html

To follow the initial intention, you would have to remove the `null` default value from `$parent`.

Reply:
any inputs on this @thirsch ?

Reply:
We thought about that in https://github.com/FriendsOfSymfony1/symfony1/pull/365#issuecomment-2041982504 and at least in theory it might be a BC if named parameters are in use, which might not be the case at all. I'm fine with both. Let me know your thoughts and I'm happy to provide a pr for it.